tests/test_test_bit_reprodubility.py

```diff
@@ -0,0 +1,293 @@
+        print(f"Test stdout: {result.stdout}\nTest stderr: {result.stderr}")
+    assert result.returncode == 0
+
+
+def test_test_bit_repro_historical_access_fail(tmp_dir):
+    """Check when checksums do not match, checksum file is still written out"""
+    test_name = "test_bit_repro_historical"
+    model_name = "access"
+
+    # Setup test Helper
+    helper = CommonTestHelper(test_name, model_name, tmp_dir)
+    helper.write_config()
+
+    # Compare checksums against the existing checksums in resources folder
+    checksum_path = helper.resources_path / "checksums" / "1-0-0.json"
+
+    # Put some expected output in the archive directory (as we are skipping
+    # the actual payu run step)
+    mock_output000 = helper.create_mock_output000()
+
+    # Modify output file to mimic a change to output file in archive
+    with (mock_output000 / "access.out").open("a") as f:
+        f.write("[chksum] test_checksum               -1")
+
+    # Build test command
+    test_cmd = (
+        f"{helper.base_test_command()} "
+        f"--checksum-path {checksum_path} "
+        f"--control-path {helper.control_path} "
+    )
+
+    # Run test in a subprocess call
+    result = subprocess.run(shlex.split(test_cmd), capture_output=True, text=True)
+
+    # Expect test to fail
+    assert result.returncode == 1
+
+    # Test when checksums aren't matched, that file are still written
+    test_checksum = helper.output_path / "checksum" / "historical-24hr-checksum.json"
+    assert test_checksum.exists()
+    content = test_checksum.read_text()
+    assert content != checksum_path.read_text()
+
+
+def test_test_bit_repro_historical_access_no_reference_checksums(tmp_dir):
+    """Check when a reference file for checksums does not exist, that
+    checksums from the output are written out"""
+    test_name = "test_bit_repro_historical"
+    model_name = "access"
+
+    # Setup test Helper
+    helper = CommonTestHelper(test_name, model_name, tmp_dir)
+    helper.write_config()
+
+    # Put some expected output in the archive directory (as we are skipping
+    # the actual payu run step)
+    helper.create_mock_output000()
+
+    # Build test command
+    test_cmd = f"{helper.base_test_command()} " f"--control-path {helper.control_path} "
+
+    # Run test in a subprocess call
+    result = subprocess.run(shlex.split(test_cmd), capture_output=True, text=True)
+
+    # Expect test to fail
+    assert result.returncode == 1
+
+    # Test that checksums are still written out
+    test_checksum = helper.output_path / "checksum" / "historical-24hr-checksum.json"
+    assert test_checksum.exists()
+
+    # Test that they are equal to checksums in resource folder
+    checksum_path = helper.resources_path / "checksums" / "1-0-0.json"
+    assert test_checksum.read_text() == checksum_path.read_text()
+
+
+def test_test_bit_repro_historical_access_no_model_output(tmp_dir):
+    """Check when a test exits, there are no checksums are in output directory
```

Comment:
```suggestion
    """Check when a test exits, that there are no checksums in the output directory
```
Just a typo. Is that what you meant?

Reply:
Yes thanks, I've added that in